support-workers/src/main/scala/com/gu/support/workers/lambdas/UpdateSupporterProductData.scala

```diff
@@ -0,0 +1,125 @@
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Digital Subscription",
+            ))
+      case SendThankYouEmailDigitalSubscriptionCorporateRedemptionState(user, product, _, subscriptionNumber) =>
+        catalogService
+          .getProductRatePlan(DigitalPack, product.billingPeriod, NoFulfilmentOptions, NoProductOptions, Corporate)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Gift Digital Subscription",
+            ))
+      case SendThankYouEmailDigitalSubscriptionGiftRedemptionState(user, product, subscriptionNumber, _) =>
+        catalogService
+          .getProductRatePlan(DigitalPack, product.billingPeriod, NoFulfilmentOptions, NoProductOptions, Gift)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Guardian Weekly",
+            ))
+      case SendThankYouEmailGuardianWeeklyState(user, product, giftRecipient, _, _, _, _, subscriptionNumber, _) =>
+        val readerType = if(giftRecipient.isDefined) Gift else Direct
+        catalogService
+          .getProductRatePlan(GuardianWeekly, product.billingPeriod, product.fulfilmentOptions, NoProductOptions, readerType)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Guardian Weekly",
+            ))
+      case SendThankYouEmailPaperState(user, product, _, _, _, _, subscriptionNumber, _) =>
+        catalogService
+          .getProductRatePlan(Paper, product.billingPeriod, product.fulfilmentOptions, product.productOptions)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = s"support-workers added ${product.productOptions.toString}",
+            ))
+      case _ => None
+    }
+
+  def supporterRatePlanItem(subscriptionName: String, identityId: String, productRatePlanId: String, productRatePlanName: String) =
+    SupporterRatePlanItem(
+      subscriptionName = subscriptionName,
+      identityId = identityId,
+      gifteeIdentityId = None,
+      productRatePlanId = productRatePlanId,
+      productRatePlanName = productRatePlanName,
+      termEndDate = LocalDate.now.plusYears(1),
```

Comment:
will this record get overwritten automatically when the query happens?

Reply:
Yes